Packs/Box/Integrations/BoxEventsCollector/BoxEventsCollector.py

```diff
@@ -0,0 +1,227 @@
+    created_after: Optional[str]
+    # validators
+    _normalize_after = validator('created_after', pre=True, allow_reuse=True)(
+        get_box_events_timestamp_format
+    )
+
+    class Config:
+        validate_assignment = True
+
+
+class BoxEventsRequest(IntegrationHTTPRequest):
+    url = parse_obj_as(AnyUrl, 'https://api.box.com/2.0/events')
+    method = Method.GET
+    params: BoxEventsParams
+
+
+class BoxEventsClient(IntegrationEventsClient):
+    request: BoxEventsRequest
+    options: IntegrationOptions
+
+    def __init__(
+        self,
+        request: BoxEventsRequest,
+        options: IntegrationOptions,
+        box_credentials: BoxCredentials,
+        session=requests.Session(),
+    ) -> None:
+        self.box_credentials = box_credentials
+        super().__init__(request, options, session)
+
+    def set_request_filter(self, after: Any):
+        self.request.params.stream_position = after
+
+    def authenticate(self):
+        request = IntegrationHTTPRequest(
+            method=Method.POST,
+            url='https://api.box.com/oauth2/token',
+            data=self._create_authorization_body(),
+            verify=self.request.verify,
+        )
+
+        response = self.call(request)
+        self.access_token = response.json()['access_token']
+        self.request.headers = {'Authorization': f'Bearer {self.access_token}'}
+
+    def _create_authorization_body(self):
+        claims = Claims(
+            client_id=self.box_credentials.boxAppSettings.clientID,
+            id=self.box_credentials.enterpriseID,
+        )
+
+        decrypted_private_key = _decrypt_private_key(
+            self.box_credentials.boxAppSettings.appAuth
+        )
+        assertion = jwt.encode(
+            payload=claims.dict(),
+            key=decrypted_private_key,
+            algorithm='RS512',
+            headers={
+                'kid': self.box_credentials.boxAppSettings.appAuth.publicKeyID
+            },
+        )
+        body = {
+            'grant_type': 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+            'assertion': assertion,
+            'client_id': self.box_credentials.boxAppSettings.clientID,
+            'client_secret': self.box_credentials.boxAppSettings.clientSecret,
+        }
+        return body
+
+
+class BoxGetEvents(IntegrationGetEvents):
+    client: BoxEventsClient
+
+    def get_last_run(self: Any) -> dict:  # type: ignore
+        demisto.debug(f'setting {self.client.request.params.stream_position=}')
+        return {'stream_position': self.client.request.params.stream_position}
+
+    def _iter_events(self):
+        self.client.authenticate()
+        demisto.debug('authenticated successfully')
+        # region First Call
+        events = self.client.call(self.client.request).json()
+        # endregion
+        # region Yield Response
+        while True:  # Run as long there are logs
+            self.client.set_request_filter(events['next_stream_position'])
```

Comment:
To what does `next_stream_position` refer?
Is it a page number of some sort? An ID of the last fetched item?

What will happen in the following scenario:
* Request for 500 events
* Received 300 (there are no more)
* **on the 3rd party, new events arrived**
* We request events again

If it refers to a page number - we might miss events.
Please document your answer in the code.

Reply:
its not refering to a page number, but to the next cursor (next stream position). It'll fetch what it can and then will try to get new ones from the next stream position.